src/components/Categories/Categories.js

```diff
@@ -0,0 +1,78 @@
+import {useEffect, useState} from 'react';
+import styled from 'styled-components';
+
+export default function Categories({menus, handleCategoryClick, curClickedCategory}) {
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    /* Get categories form the given menus */
+    function getCategories(menus) {
+      const categories = menus.reduce(
+        function (pre, cur) {
+          if (!pre.includes(cur.category)) {
+            pre.push(cur.category);
+          }
+          return pre;
+        },
+        ['All']
+      );
+      return categories;
+    }
+
+    setCategories(getCategories(menus));
+
+    /* Sort categories in this order */
+    const sortBy = ['All', 'Starters', 'Mains', 'Pastries', 'Drinks'];
+    const categorySort = ({data, sortBy, sortfield}) => {
+      const sortByObj = sortBy.reduce(
```

Comment:
Same as above.

Reply:
I can do render without this function, but the menus will be rendered in different order, like Mains, Starters.....  So this is the solution to the problem.

Reply:
see my solution above

Reply:
Thank you, works good in my app.